### PR TITLE
[www] WIP: Replace Tex Gyre Schola with Spectral + …more

### PR DESCRIPTION
**Pull request by fk**
Sloppy WIP…

- Replace TeX Gyre Schola with Spectral … again. I'm still undecided, still convinced it's spaced a little too loose. If I only had to choose based on rendering and italic though … ;-). FWIW, Tex Gyre Schola is still not removed, so you can see how Gyre Schola looks at 1.5 line-height by changing `bodyFontFamily` in src/utils/typography back to `Tex Gyre Schola`. (Also there’s a Typekit script included that loads Garamond, PT Schoolbook and Calluna to play around with.)
- Bump baseLineHeight to 1.5 (was 1.4), blockMarginBottom to 0.8 (was 0.65); adjust a whole bunch of relatively sized components accordingly.
- Add previous/next links to blog posts.
- Add canonicalLink and publishedAt fields to blog post frontmatter. Probably should be canonical_link and published_at?!
- If a blog post has a canonicalLink, this adds a link rel="canonical" to the document head; in the meantime I discovered https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-canonical-urls though…
- Add CSS class `post-body` to the blog post container.
- When in .post-body, pull .gatsby-highlight and .gatsby-resp-iframe-wrapper into the horizontal padding of their container (just like .gatsby-resp-image-link).
- Links in blog post copy are now Futura PT bold (currently only there).
- headerColor and bodyColor are now derived from the brand color.
- Reduce scaleRation to 2 (was 2.15), h1 font-size.
- No more italic h3.
- Don’t bump main navigation social link font-size for Desktop and up.
- Handle border around .gatsby-highlight via boxShadow.
- Simplify .gatsby-highlight-code-line styles, use rhythm(3/4) instead of blockMarginBottom.
- Add a bunch of new article images, some unused. Not super happy about the ones used in articles, either. All licensed under CC.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Quite the changelog! :-)

https://deploy-preview-1765--gatsbyjs.netlify.com/blog/gatsbygram-case-study/

I'd vote the current line-height is a bit loose. I'd say go down to at least 1.45.

**Comment by KyleAMathews** [comment]
Those bold futura links are *beautiful* ❤️

**Comment by KyleAMathews** [comment]
Got to get the image text stuff out of the excerpt text.

screen shot 2017-08-10 at 9 57 20 pm

**Comment by fk** [comment]
> Quite the changelog! :-)

Yeah, horrible … procrastination 💯 , typing all that crap out is the least I could do.
It's all quite easy to rip apart though 😛

File: www/src/templates/template-blog-post.js [hunk 1]
```diff
@@ -1,170 +1,282 @@
...
+    console.log(headLinks)
     return (
-      <Container>
-        {/* Add long list of social meta tags */}
-        <Helmet
-          title={post.frontmatter.title}
-          link={[
-            {
-              rel: `author`,
-              href: `https://gatsbyjs.org${post.frontmatter.author.slug}`,
```
Comment (fk) [inline www/src/templates/template-blog-post.js]: Just noticed this, seems author pages were planned already?

Comment (KyleAMathews) [inline www/src/templates/template-blog-post.js]: Yeah that was the idea just I was the only author for a while :-)

Comment (fk) [inline www/src/templates/template-blog-post.js]: Thinking about how the author page would look/how author bios would look on blog post pages once we have a dedicated author page in place…

Should we just crop the author bio on the blog post page or distinguish between "short bio" and "author page bio" and offer fields for both?

I saw that there's already field for Twitter in authors.yaml already so we can at least show a Twitter icon on the author page. Maybe we should add some fields for the blog info (URL, title)? What about an (optional) link to a Github profile, too?

**Comment by fk** [comment]
> Got to get the image text stuff out of the excerpt text.

Let's pull the image caption/credit into frontmatter I guess, since the main article image is already there anyway? Separately quering/implementing that image also enables more design/layout options and might be a little more "correct" in terms of semantics/design?

Hm. Wondering about the RSS feed now. :-D

File: www/src/templates/template-blog-post.js (hunk 1 above)
Comment (KyleAMathews) [inline www/src/templates/template-blog-post.js]: Yeah, that all sounds great!

**Comment by KyleAMathews** [comment]
>Let's pull the image caption/credit into frontmatter I guess

Yeah, that'd work. Better than adding heuristics to the "excerpt" code in gatsby-transformer-remark. Plus yeah, then we could add the image Medium-style to the blog indexes.

File: www/src/templates/template-blog-post.js (hunk 1 above)
Comment (fk) [inline www/src/templates/template-blog-post.js]: 👍 Let's see if I can manage to create those pages then :)
URLs like https://www.gatsbyjs.org/blog/author/kyle-mathews?

Comment (KyleAMathews) [inline www/src/templates/template-blog-post.js]: how about just gatsbyjs.org/contributors/kyle-mathews

File: www/gatsby-node.js [hunk 2]
```diff
@@ -35,20 +44,42 @@ exports.createPages = ({ graphql, boundActionCreators }) => {
...
+
+            if (_.includes(slug, `/blog/`) && !draft) {
+              return edge
+            }
+          }
+        )
+
+        // Create blog pages.
+        blogPosts.forEach((edge, index) => {
+          const next = index === 0 ? false : blogPosts[index - 1].node
```
Comment (KyleAMathews) [inline www/gatsby-node.js]: you can just query for this stuff directly btw :-) `allMarkdownRemark { edges { next { id }, prev { id }, node { id }}}`

Gotta get the GraphQL docs written sigh...

File: www/src/html.js [hunk 3]
```diff
@@ -72,6 +72,12 @@ export default class HTML extends React.Component {
             href={`/safari-pinned-tab.svg`}
             color="#5bbad5"
           />
+          <script src="https://use.typekit.net/vol0sfl.js" />
```
Comment (KyleAMathews) [inline www/src/html.js]: What's this back for?

File: www/src/utils/typography.js [hunk 4]
```diff
@@ -60,17 +73,16 @@ const options = {
         paddingBottom: `0.1em`,
       },
       ".gatsby-highlight": {
-        // background: `hsla(23, 60%, 97%, 1)`,
         background: colors.a[0],
-        border: `1px solid ${colors.a[1]}`,
+        boxShadow: `inset 0 0 0 1px ${colors.a[1]}`,
         borderRadius: `${presets.radius}px`,
-        padding: rhythm(options.blockMarginBottom),
```
Comment (KyleAMathews) [inline www/src/utils/typography.js]: Curious why the change here?

File: www/src/html.js (hunk 3 above)
Comment (fk) [inline www/src/html.js]: Was playing around with some serifs from Typekit again; see first bullet point in the PR description. Gyre Schola is also still in.

Comment (KyleAMathews) [inline www/src/html.js]: 👍

Comment (fk) [inline www/src/html.js]: Removed!

File: www/src/utils/typography.js (hunk 4 above)
Comment (fk) [inline www/src/utils/typography.js]: I was playing around with pulling the highlighted code blocks into the containers horizontal padding and wanted things to line up. Not too strongly attached to this either btw :D

image

Comment (KyleAMathews) [inline www/src/utils/typography.js]: Oooo hadn't noticed that detail. I likey!

**Comment by fk** [comment]
Thanks for "greenkeeping this PR", @KyleAMathews. Last three days have been (unexpectedly) crazy busy, sorry for not following up more recently. Will try to wrap this up tomorrow!

**Comment by KyleAMathews** [comment]
So on the 0.75 stuff for paddings. Dunno why it hadn't occurred to me before but we should just do the side padding with `blockMarginBottom` as well so then everything is aligned.

Also... 1.45 for line-height *still* feels a bit too loose. How's 1.4 look to ya? Also maybe drop the `blockMarginBottom` a bit too?

**Comment by KyleAMathews** [comment]
And no worries about being busy!

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit da7a3ae1de8c8e41cdfc7942b27362dc3218f244

https://app.netlify.com/sites/using-glamor/deploys/599f2295a700c451a8baa69e

File: www/src/templates/template-blog-post.js [hunk 5]
```diff
@@ -24,118 +46,272 @@ const BlogPostTemplate = React.createClass({
...
+            <title>
+              {post.frontmatter.title}
+            </title>
+            <link
+              rel="author"
+              href={`https://gatsbyjs.org${post.frontmatter.author.slug}`}
+            />
+            {post.frontmatter.canonicalLink &&
+              <link rel="canonical" href={post.frontmatter.canonicalLink} />}
+            <meta
```
Comment (fk) [inline www/src/templates/template-blog-post.js]: One of the things this horrible PR does is based on the presence of `canonicalLink` and `publishedAt` in frontmatter, output "Originally posted at ${publishedAt}" with the href pointing to `canonicalLink` in the blog post header.

I thought it'd be smart to use that information to output a `<link rel="canonical">` and added that along the way. Then through inspecting the source, I found out about https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-canonical-urls …

* Should I remove the additional <link>?
* camelCase is OK for fields in frontmatter?

File: www/src/templates/template-blog-post.js [hunk 6]
```diff
@@ -24,118 +46,272 @@ const BlogPostTemplate = React.createClass({
...
+          <Helmet>
+            <title>
+              {post.frontmatter.title}
+            </title>
+            <link
+              rel="author"
+              href={`https://gatsbyjs.org${post.frontmatter.author.slug}`}
+            />
+            {post.frontmatter.canonicalLink &&
+              <link rel="canonical" href={post.frontmatter.canonicalLink} />}
```
Comment (fk) [inline www/src/templates/template-blog-post.js]: One of the things this horrible PR does is based on the presence of `canonicalLink` and `publishedAt in` frontmatter, output "Originally posted at ${publishedAt}" with the href pointing to `canonicalLink` in the blog post header.

I thought it'd be smart to use that information to output a `<link rel="canonical">` and added that along the way. Then through inspecting the source, I found out about https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-canonical-urls …

* Should I remove the additional ?
* camelCase is OK for fields in frontmatter?

Comment (KyleAMathews) [reply www/src/templates/template-blog-post.js]: It definitely makes sense to have the canonical link point to the other site. But this way would add two links which is confusing :-)

Perhaps the other plugin could expose an API for overriding the default canonical link or something.

But another excellent idea is to just drop it in interest of making it easier to land this PR :-)

Comment (fk) [reply www/src/templates/template-blog-post.js]: ;) :D – done!

**Comment by fk** [comment]
My time during the weekend is quite fragmented, so I apologize in advance for causing noise serving the updates to and the remaining questions regarding this PR bit-by-bit … as you said, let's wrap this up … before it has its one-month anniversary. ;-)

Cozy sidebar links and headlines, reduced headline sizes (via `scaleRatio`) and a tad smaller main navigation logo and font-size (before/after):

image

**Comment by fk** [comment]
I didn't go very far working on the blog index, but went back to the one-line author-and-date variant I already had in some PR earlier. Here's why: I don't see a reason why this information needs to occupy more then one line, really ;-) – personally it distracts me a little when "scanning" the page. We also chatted about blog posts with more then one author, and I feel the one-line solution is a little more future-proof/-ready for that, too.

Here's a screenshot from the great Segment blog showing multiple authors:

image

I think the Segment blog does a great job and would suggest moving Gatsby to something quite similar to their layout for both index and single post pages. I guess now it's obvious why we also have fancier, more readable dates now:

image

**Comment by fk** [comment]
Also in are the contributor pages, currently sporting a very simple layout:

image

I haven't thought about what additional fields to add to [author.yaml](https://github.com/gatsbyjs/gatsby/blob/master/docs/blog/author.yaml) since we initially talked about it. I believe we discussed

* separate fields for short and long bio
* adding a field for blog and/or homepage (so we can remove that kind of info from the bio: 
image)
* adding a field for Github

– what's your take on that? Once we are settled on those, I'll propose an updated layout; can't think of more than some icons added to the links though, maybe explore putting the blog posts in cards and showing them on a background color similar to the sidebar, … maybe exploring a "split-screen layout" (2 columns, left author image, bio, and links, and blog posts to the right)?

**Comment by fk** [comment]
Also added more whitespace above and below remark-images and their caption…

image

… and made the highlighted code block background- and border-color a bit brighter/calmer and increased font-size a little, all which IMO improves readability a tad; code blocks are also pulled into the blog post margins, and for large screens their padding is increased a little, too:

image

**Comment by fk** [comment]
On smaller screens, code blocks are staying pulled into the post margins, but border-radius and vertical borders are dropped:

image

Still gotta make that work nicely for when we are displaying a sidebar:

image

**Comment by KyleAMathews** [comment]
Everything is looking fantastic! Lemme know when you're satisfied enough to make the release :-D

**Comment by fk** [comment]
Good to go! 🎉 Will follow up with cards for blog posts in a separate PR.

**Comment by KyleAMathews** [comment]
🎉 yeah! Excited to get this out there!

**Comment by KyleAMathews** [comment]
Thanks again for your fantastic work here!
